snapcraft/commands/login.py

```diff
@@ -47,7 +47,8 @@ def main(argv=None):
     print('Enter your Ubuntu One SSO credentials.')
     email = input('Email: ')
     password = getpass.getpass('Password: ')
-    otp = input('OTP: ')
+    otp = input('One-time password (just press enter if you don\'t use '
+                'two-factor authentication): ')
```

Comment:
It's a bit verbose as we discussed, but I think it's a good intermediate solution until the uploader is smart enough to know whether or not it's actually required.